protobuf/v1/replica.proto

```diff
@@ -96,3 +98,29 @@ message CreateReplicaSnapshotRequest {
 message CreateReplicaSnapshotResponse {
   string snapshot_uuid = 1; // UUID of the snapshot newly created
 }
+
+/// List Replica Snapshot Request.
+message ListReplicaSnapshotsRequest {
+  string replica_uuid = 1;  // uuid of the replica to list snapshots for
+}
+
+/// Replica Snapshot Params to Return in ListSnapshot.
+message ReplicaSnapshot {
+  string snapshot_uuid = 1;    // uuid of the snapshot
+  string snapshot_name = 2;    // name of the snapshot
+  string replica_uuid = 3;     // uuid of the replica this snapshot is taken from
+  uint64 bytes_referenced = 4; // amount of bytes referenced by snapshot
```

Comment:
what does this "referenced" actually mean?
Does `bytes_referenced ` grow as we write to the volume?

Reply:
> what does this "referenced" actually mean? Does `bytes_referenced ` grow as we write to the volume?

As per the current understanding from SPDK, this value will be fixed for the snapshot as snapshot will be a LVOL which is readonly. So the value will not be changed.

Reply:
So what is the bytes referring to then? The size of the original replica?

Then we might want to have another size, which reflects how much additional storage is taken by the snapshot, as in, if we write to the replica, we're actually allocating more storage right? Would be good to report how much storage this snapshot is "costing" us ?
Also makes me wonder, in case of snapshot, will sizes reported by "ReplicaSpaceUsage" change?